ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -572,6 +574,12 @@ public void delete(List<MessageInfo> infosToDelete) throws StoreException {
           indexValuesToDelete.add(value);
           lifeVersions.add(info.getLifeVersion());
         }
+        if (!value.isDelete() && !value.isUndelete()) {
```

Comment:
Why TtlUpdate is also counted here?

Reply:
Because we use findKey to get the value, and in findKey, we are not returning TTL_UPDATE record.

Reply:
because then the TTL_UPDATE flag is true, value.isPut will be false. findKey has this weird logic where it will populate TTL_UPDATE flag even if we are not trying to find TTL_UPDATE record. 

So when we have PUT, TTL_UPDATE and DELETE for this blob and we call findKey(blobId, [INDEX_TYPE.PUT]),  we will get a PUT value with TTL_UPDATE flag being true. And if we do value.isPut, it will return false.

Reply:
I see. Thanks for explanation. I think it probably justifies a comment here.